wiki.moinmoin

```diff
@@ -91,216 +93,423 @@ Notes:
+RULE:   to use the Ubuntu archive version. When this is not possible, the
+RULE:   security team must agree to using the embedded source.
+RULE: - Similarly, when a binary from one source package statically links to
+RULE:   libraries from another source package from the archive, when those
+RULE:   libraries are updated the statically linked binaries must be rebuilt
+RULE:   with the updated libraries to receive the fix, which increases the
+RULE:   maintenance burden. For this reason, static linking in archive builds
+RULE:   is discouraged unless static linking is required for the package in
+RULE:   question to function correctly (e.g. an integrity scanner).
+RULE: - Does debian/control use `Built-Using`? This may indicate static linking
+RULE:   which should be discouraged (except golang, see below)
+
 OK:
 TODO: - no embedded source present
 TODO: - no static linking
+TODO: - does not have odd Built-Using entries
+
+RULE: Golang
+RULE: - golang 1.4 packages and earlier could only statically compile their
+RULE:   binaries. golang 1.5 in Ubuntu 16.10 introduced `-buildmode=shared`
+RULE:   to build shared libraries and `-linkshared` to dynamically link against
+RULE:   shared libraries. In general, statically compiled binaries are not
+RULE:   suitable for the Ubuntu archive because they increase the maintenance
+RULE:   burden significantly. As such, from Ubuntu 16.10 and later, golang
+RULE:   packages packages in main were expected to be built with shared
+RULE:   libraries.
+RULE: - Evaluating cost/benefits while considering the ABI instability of golang
+RULE:   libraries during this period, the MIR team decided for 17.10 and later
+RULE:   to allow static builds of golang packages in main, so long as the number
+RULE:   of these packages remains low and they follow the guidelines below:
+RULE:   - golang applications in main are expected:
+RULE:       1. to build using `golang-*-dev` packages from the Ubuntu archive
+RULE:          with `Built-Using` in debian/control. This requirement ensures
+RULE:          that the security team is able to track security issues for all
+RULE:          affected static binary packages
+RULE:       2. not to build any vendored (i.e. embedded) code in the source
+RULE:          package whose binaries appear in the archive (e.g. test code is
+RULE:          ok) without clear justification from the requesting team and
+RULE:          approval from the security team. This requirement ensures that
+RULE:          the security team is able to track security issues for all
+RULE:          affected source packages.
+RULE:       3. only build against approved vendored sources (when applicable)
+RULE:          and not build against newly added vendored code in subsequent
+RULE:          Ubuntu uploads without approval from the security team. This
+RULE:          requirement ensures that the security team is able to track
+RULE:          security issues for all affected source packages.
```

Comment:
This part is a little bit tricky, because you may intend that you never update your vendored dependencies. This would:
- introduce new code, with increased complexity
- those dependencies themselves may have new dependencies, and so, this would block it.

Also, I didn’t find from the reportee side a similar explanation than the reviewer one.

Maybe we should revisit this?